### ISSUE TITLE
Authors of unapproved versions

### ISSUE THREAD

**Issue by Dr-Yukon** [cosmocode/dokuwiki-plugin-publish#142]
Please add a column for observing authors of last edits

**Comment by Rainbow-Spike** [comment]
Please!

**Comment by Chris75forumname** [comment]
I need exactly that, too: A line with the author of last the edit in each banner.

Since this is already shown at the bottom of the page (page footer), it should be not so hard to transfer into the banner (or page header). I will try to build it in. I will post my solution, if i succeed.

Yours, Christian

**Comment by Chris75forumname** [comment]
Hi Spike,
I managed to do so! It was not all that hard to do. It took me some time only because I am a total beginner. I just modified the messages in the banner. (I think, columns would be great too, it should be possible, but I didn't try to make columns and I will stick with the banner layout right now.)
I found some code how to display the last editor here: https://www.dokuwiki.org/plugin:verinfo (an old plugin, but the code is plain and well documented). Also I found the layout of the $INFO array here: https://www.dokuwiki.org/devel:infoarray (so I could verify I use the correct data). I will send you all the necessary changes / files, if you want? (I run latest Dokuwiki Hogfather and latest version of publish plugin.)
Yours, Christian

**Comment by Rainbow-Spike** [comment]
How can I test it?

**Comment by Chris75forumname** [comment]
Alright, there is one little thing you would have to do, after I send you all the files, I recommend using: https://www.deepl.com/translator to translate all my german changes into your language. (some little bits of language are in the code of banner.php) This is just my quick and dirty fix, so I didn't know how to do this better.
And please, make safety copies of all files you overwrite with mine.
By the way, I found I had to set my timezone in order to have the right timestamps in the publish plugin by setting it in conf/local.protected.php https://www.dokuwiki.org/start?id=tips:timezone
Goto: dokuwiki/lib/plugins/publish     and: .../publish/action    and: .../publish/lang   and replace the files given.
[files.zip](https://github.com/cosmocode/dokuwiki-plugin-publish/files/6874508/files.zip)
Yours, Christian

**Comment by Rainbow-Spike** [comment]
Christian, I already use DeepL %)
in November 2020 one German came to my project and we communicate using this translator (...and also a Frenchman, a Spaniard, a Pole, and a couple of Hungarians. We're a warm bunch). BTW, this German got the German translation of a great sci-fi webcomic halfway through yesterday and it got a silver star ;)

**Comment by Rainbow-Spike** [comment]
Also I use a userscript to manage edit time and plan to eventually make it into a plugin with an option in the profile and storing the time zone in localStorage. And we have a huge variety of backups ≺(☉‿☉)≻

**Comment by Chris75forumname** [comment]
Great! I used DeepL last year to translate a book (150 p.) from Dutch to German even though I do not speak a single word of Dutch. It worked! :-)

**Comment by Chris75forumname** [comment]
Wow! What do you need this for? If you collaborate with others all around the world?

**Comment by Rainbow-Spike** [comment]
Yes, it's more effective than GT, and even than Yandex Translator with Tomita-parser. The convolutional neural network win!

**Comment by Chris75forumname** [comment]
No, I meant your timefix plugin? :-)

**Comment by Rainbow-Spike** [comment]
I live in Russian Far East +10 GMT Vladivostok
server use GMT+0 London
most of my readers so far are from GMT+3 Moskow
and quite a few readers from other zones

**Comment by Chris75forumname** [comment]
Wow! Great! I love Russia! Alright, let me know if it works and I will try to help if anything does not work. Yours, Christian

**Comment by Rainbow-Spike** [comment]
And conf/local.protected.php causes the Discord RSS-bot to revolt, so I don't use it...

**Comment by Chris75forumname** [comment]
Oh, I got it!

**Comment by Rainbow-Spike** [comment]
Please take a look at our work https://comicslate.org/de/sci-fi/freefall/d0000

**Comment by Chris75forumname** [comment]
Wow! Cool! Great work!!!

**Comment by Rainbow-Spike** [comment]
You can help us if you want ≺(☉‿‿☉)≻

**Comment by Chris75forumname** [comment]
Thank you for your trust in me! I am afraid I am too busy with other things! Yours, Christian

**Comment by Rainbow-Spike** [comment]
It works on Greebo!
http://screenshot.ru/3c72fbae852f11266e93f30ee93ee284.png =>
http://screenshot.ru/3c6b6079a134409e565ddf6a60f16aaa.png

**Comment by Rainbow-Spike** [comment]
But we don't use the Discussion plugin, instead we use the Discord conference

**Comment by Chris75forumname** [comment]
Oh, I forgot this feature. I added a note at the end of the code, because the built-in internal_note function did not work as expected.
This is easy to fix, just comment out additionally two of all the following lines, so all of these lines are commented out in the function showDraft() in banner.php:
  `echo '<br/>';
//      printf(' ' . $this->getLang('approvals'), $approvalCount, $this->getConf('number_of_approved'));     // ++++++ Diese Zeile wird nicht angezeigt! ++++++
        echo $this->getLang('approve_note');                                                     // ++++++ Hinzugefügt weil showInternalNote() nicht funktioniert! ++++++
//      if ($approvalCount != 0) {
//          printf(' ' . $this->getLang('approved by'), implode(', ', $this->hlp->getApprovers()));
//      }`

**Comment by Chris75forumname** [comment]
If you want your number_of_approved shown as in your old banner, you can uncomment the following two lines again:
`echo '<br/>';`
`printf(' ' . $this->getLang('approvals'), $approvalCount, $this->getConf('number_of_approved'));`

**Comment by Rainbow-Spike** [comment]
http://screenshot.ru/16d1096d83a35224bb0a5a00ba99feac.png nice

**Comment by Chris75forumname** [comment]
BTW, all comments like ++++ comment ++++ were added by me. Comments without ++++ are original comments.

**Comment by splitbrain** [comment]
Guys, please stay on topic here.

@Chris75forumname it would be helpful if you could send a proper pull request.

**Comment by Rainbow-Spike** [comment]
> Guys, please stay on topic here.
Sorry, there is no other way to keep contact here on GitHub, even PMs is absent. Even the Discord server "Dokuwiki Community" is not even open yet ;)

**Comment by Chris75forumname** [comment]
Hi Andi, 
you are right! I wish I could, but I am afraid, sending a proper pull request is beyond my abilities! All I can do is speak a decent English and do some really nasty hacking. I am sorry! 

Hi Spike,
Today, my colleges started testing my tweak by creating pages like wild! Which sent me on a bug hunt myself all day. The problem was, when a page is freshly created, the contributor array is NOT generated right away. Only after the next edit. Since my colleges were using the nice little manual I wrote, all went well and next edits were not necessary. So most of the new pages were shown with NO last editor/creator in the banner. Which did not look appealing! So I added an if-else construct to inquire if the contributor array does exist and if so, show the last editor, and if not, to fall back to the creator of the page, each provided as full user names. 
So this is the new version of my tweak: 
[banner.php.zip](https://github.com/cosmocode/dokuwiki-plugin-publish/files/6880416/banner.php.zip)
I did not have time to fully validate it and test it. And I won't have time to do so in the next 4 weeks from now. So, please use with caution! I would be grateful for tester's feedback! 

Thanks to you all!
Christian

**Comment by Chris75forumname** [comment]
> Since this is already shown at the bottom of the page (page footer), it should be not so hard to transfer it i

BTW: Where is actually the code that generates the pagefooter? I couldn't find it anywhere.

**Comment by Rainbow-Spike** [comment]
look at /lib/tpl/your_template/ - files named main.php or footer.html

**Comment by Chris75forumname** [comment]
No, I mean the part of the code inside dokuwiki, which uses `$lang['lastmod']` to display last modification of the current page in the page footer. None of these plugins do anything like that. I should figure out how to do a text search for this string across all php files in the entire dokuwiki directory. Hm? Cheers!

**Comment by Chris75forumname** [comment]
Hi, I found a simple way to search text strings inside all php files of my /dokuwiki directory:
Open Windows terminal, change to your dokuwiki directory
`cd C:\dokuwiki`
execute the following command with your "textstring" in place:
`findstr /L /S textstring *.php`
This way I found what I was looking for. The pagefooter "lastmod" message is created by the function `tpl_pageinfo()` in `dokuwiki/inc/template.php` ( /** Print some info about the current page **/ ).
This uses the function `editorinfo()` from `dokuwiki/inc/common.php`.
After checking all these out, I still think my solution is the best. So nothing is changed! It seems to work as expected, now!
Enjoy!
